altair/vegalite/v5/schema/channels.py

```diff
@@ -72,7 +72,7 @@ def to_dict(
                 # We still parse it out of the shorthand, but drop it here.
                 parsed.pop("type", None)
             elif not (type_in_shorthand or type_defined_explicitly):
-                if isinstance(context.get("data", None), pd.DataFrame):
+                if is_pandas_dataframe(context.get("data", None)):
```

Comment:
```suggestion
                if _is_pandas_dataframe(context.get("data", None)):
```
Avoiding adding to `__all__` (3/3)